src/gui/sdl_mapper.cpp

```diff
@@ -404,14 +412,13 @@ class CKeyBind final : public CBind {
 		return sdl_scancode_name;
 	}
 
-	bool HasSameBinding(const CKeyBind* other) const
-	{
-		return other && key == other->key && mods == other->mods;
-	}
-
-	const char* GetEventName() const
+	bool HasSameBinding(const CBind* other) const override
 	{
-		return event ? event->GetName() : "";
+		const auto other_typed = dynamic_cast<const CKeyBind*>(other);
+		if (!other_typed) {
+			return false;
+		}
+		return other_typed && key == other_typed->key && mods == other->mods;
```

Comment:
You're checking whether `other_typed` is null twice.  You could remove the if statement and it'll work exactly the same.  Alternatively, you could say you're leaving the separate if statement for readability and remove the `other_typed &&` from the return statement.